gearcmd/worker_test.go

```diff
@@ -202,3 +205,44 @@ func TestHaltGraceful(t *testing.T) {
 	_, err := config.Process(mockJob)
 	assert.NoError(t, err)
 }
+
+func TestProcessWithErrorBackoff(t *testing.T) {
+	// Get a temp file for the script to hold its state.
+	file, err := ioutil.TempFile("", "temp")
+	assert.NoError(t, err)
+	filename := file.Name()
+	defer os.Remove(filename)
+	defer file.Close()
+	mockJob := mock.CreateMockJob(filename)
+	config := TaskConfig{
+		FunctionName:            "name",
+		FunctionCmd:             "testscripts/nonZeroExit.sh",
+		LastResults:             ring.New(2),
+		ErrorResultsBackoffRate: 10 * time.Millisecond,
+	}
+	mockClock := clock.NewMock()
+	gearcmdconfig.Clock = mockClock
+	defer func() {
+		gearcmdconfig.Clock = clock.New()
+	}()
+
+	assert.Equal(t, config.errorResultsBackoff, 0)
+	response, err := config.ProcessWithErrorBackoff(mockJob)
+	assert.Nil(t, response)
+	assert.EqualError(t, err, "exit status 2")
+	assert.Equal(t, config.errorResultsBackoff, config.ErrorResultsBackoffRate)
+
+	done := make(chan bool)
+	go func() {
+		response, err := config.ProcessWithErrorBackoff(mockJob)
+		assert.Nil(t, response)
+		assert.EqualError(t, err, "exit status 2")
+		done <- true
+	}()
+	time.Sleep(100 * time.Millisecond)
```

Comment:
is this just time for the shell script to execute? can you comment w/ what it's for?

Reply:
Yeah I couldn't figure out a way to block the main test thread without sleeping. If you know a better way let me know.